apps/api/src/domain/models/audio-visual/audio-item/repositories/arango-audio-item-query-repository.ts

```diff
@@ -0,0 +1,119 @@
+import { AggregateId } from '../../../../../domain/types/AggregateId';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { Maybe } from '../../../../../lib/types/maybe';
+import { isNotFound } from '../../../../../lib/types/not-found';
+import { ArangoConnectionProvider } from '../../../../../persistence/database/arango-connection.provider';
+import { ArangoDatabase } from '../../../../../persistence/database/arango-database';
+import { ArangoDatabaseForCollection } from '../../../../../persistence/database/arango-database-for-collection';
+import mapDatabaseDocumentToAggregateDTO from '../../../../../persistence/database/utilities/mapDatabaseDocumentToAggregateDTO';
+import mapEntityDTOToDatabaseDocument from '../../../../../persistence/database/utilities/mapEntityDTOToDatabaseDocument';
+import { IAudioItemQueryRepository } from '../queries/audio-item-query-repository.interface';
+
+export class ArangoAudioItemQueryRepository implements IAudioItemQueryRepository {
+    private readonly database: ArangoDatabaseForCollection<
+        IDetailQueryResult<IAudioItemViewModel & { actions: ICommandFormAndLabels[] }>
+    >;
+
+    constructor(arangoConnectionProvider: ArangoConnectionProvider) {
+        this.database = new ArangoDatabaseForCollection(
+            new ArangoDatabase(arangoConnectionProvider.getConnection()),
+            'audioItem__VIEWS'
+        );
+    }
+
+    async create(
+        view: IDetailQueryResult<IAudioItemViewModel> & { _actions: ICommandFormAndLabels[] }
+    ): Promise<void> {
+        await this.database.create(mapEntityDTOToDatabaseDocument(view));
+    }
+
+    async createMany(
+        views: (IAudioItemViewModel & { actions: ICommandFormAndLabels[] } & {
+            actions: ICommandFormAndLabels[];
+        })[]
+    ): Promise<void> {
+        await this.database.createMany(views.map(mapEntityDTOToDatabaseDocument));
+    }
+
+    async delete(id: AggregateId): Promise<void> {
+        return this.database.delete(id);
+    }
+
+    async publish(id: AggregateId): Promise<void> {
+        await this.database.update(id, {
+            isPublished: true,
+        });
+    }
+
+    async fetchById(
+        id: AggregateId
+    ): Promise<Maybe<IAudioItemViewModel & { actions: ICommandFormAndLabels[] }>> {
+        const result = await this.database.fetchById(id);
+
+        if (isNotFound(result)) return result;
+
+        // should we rename this helper method?
+        return mapDatabaseDocumentToAggregateDTO(result) as IAudioItemViewModel & {
+            actions: ICommandFormAndLabels[];
+        };
+    }
+
+    async fetchMany(): Promise<(IAudioItemViewModel & { actions: ICommandFormAndLabels[] })[]> {
+        const documents = await this.database.fetchMany();
+
+        return documents.map(mapDatabaseDocumentToAggregateDTO) as (IAudioItemViewModel & {
+            actions: ICommandFormAndLabels[];
+        })[];
+    }
+
+    async translateName(
```

Comment:
I'm liking this optimized per use case approach. It's interesting to compare this to generic repository approach.

Reply:
cool to see the aql in use here.  looking forward to novel graph queries